packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -60,6 +60,7 @@ class PinCodeBloc extends $PinCodeBloc {
   final bool autoPromptBiometric;
 
   final BehaviorSubject<String> _pinCode = BehaviorSubject.seeded('');
+  final BehaviorSubject<bool> isAuthenticated = BehaviorSubject.seeded(false);
```

Comment:
Do we really need to introduce a new subject for authentication tracking ? 

I believe here we have a fundamental issue. The `digitsCount` state should represent the `_pinCode` length, while `_$addDigitEvent` and `_$deleteDigitEvent` should control the `_pinCode` value.